pkg/sql/catalog/catprivilege/default_privilege_test.go

```diff
@@ -637,9 +637,70 @@ func TestModifyDefaultDefaultPrivilegesForPublic(t *testing.T) {
 		descpb.DefaultPrivilegesRole{Role: creatorUser},
 		privilege.List{privilege.USAGE},
 		[]security.SQLUsername{security.PublicRoleName()},
-		tree.Types,
+		tree.Types, false,
```

Comment:
Ah one more thing, I'd like to see some tests for this where we're granting `WITH GRANT OPTION` as well